File: app/src/icloud/java/at/bitfire/davdroid/ui/ICloudAccountsDrawerHandler.java
package at.bitfire.davdroid.ui;

import android.app.Activity;
import android.content.Intent;
import android.support.annotation.NonNull;
import android.view.MenuItem;

import at.bitfire.davdroid.Constants;
import at.bitfire.davdroid.R;

public class ICloudAccountsDrawerHandler implements IAccountsDrawerHandler {

    @Override
    public boolean onNavigationItemSelected(@NonNull Activity activity, MenuItem item) {
        switch (item.getItemId()) {
            case R.id.nav_app_license:
                activity.startActivity(new Intent(activity, SubscriptionActivity.class));
                break;
            case R.id.nav_app_settings:
                activity.startActivity(new Intent(activity, AppSettingsActivity.class));
                break;
            case R.id.nav_about:
                activity.startActivity(new Intent(activity, AboutActivity.class));
                break;
            case R.id.nav_website:
                activity.startActivity(new Intent(Intent.ACTION_VIEW, Constants.webUri));
                break;
            default:
                return false;
        }

        return true;
    }

}

File: app/src/icloud/java/at/bitfire/davdroid/ui/ICloudAccountsDrawerHandler.kt
package at.bitfire.davdroid.ui;

import android.app.Activity
import android.content.Intent
import android.net.Uri
import android.view.MenuItem
import at.bitfire.davdroid.R

class ICloudAccountsDrawerHandler: IAccountsDrawerHandler {

    override fun onNavigationItemSelected(activity: Activity, item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.nav_app_license ->
                activity.startActivity(Intent(activity, SubscriptionActivity::class.java))
            R.id.nav_app_settings ->
                activity.startActivity(Intent(activity, AppSettingsActivity::class.java))
            R.id.nav_about ->
                activity.startActivity(Intent(activity, AboutActivity::class.java))
            R.id.nav_faq ->
                activity.startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(activity.getString(R.string.navigation_drawer_faq_url))))
            R.id.nav_website ->
                activity.startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(activity.getString(R.string.homepage_url))))
            else ->
                return false
        }

        return true
    }

}
